pkgs/gnolang/preprocess.go

```diff
@@ -1626,7 +1626,19 @@ func Preprocess(store Store, ctx BlockNode, n Node) Node {
 					n.Depth = depth
 					n.BodyIndex = index
 				case FALLTHROUGH:
-					// TODO CHALLENGE implement fallthrough
+					if swchC, ok := last.(*SwitchClauseStmt); ok {
+						// last is a switch clause, find its index in the switch and assign
+						// it to the fallthrough node BodyIndex. This will be used at
+						// runtime to determine the next switch clause to run.
+						swch := lastSwitch(ns)
```

Comment:
Just a question here - what is the situation when `lastSwitch` can return `nil`? (is this even possible at  this point)

If it is, then the case is not handled correctly, and can cause a panic

Reply:
Given that : 
- we have just asserted that `last` is `SwitchClauseStmt`
- `ns` contains the hierarchy of nodes of the node `n`
- `ns[0]` is the root node (`FileNode`) 
- `ns[len(ns)-1]`  is the first parent, so `SwitchClauseStmt` (so is equal to `last`).

If we can't find a `SwitchStmt` in `ns`, that means the code looks like this : 

```go
package main

func main() {
  case 0:
    fallthrough
}
```

A `case` w/o a `switch` doesn't even compile, so the preprocess step is not even executed (the execution is interrupted before). That's why I don't think it's necessary to assert that `lastSwitch` doesn't return a `nil` value.

That said I'm not against replacing `return nil` in `lastSwitch` by a `panic`, if you prefer, because maybe my assumptions can be wrong in some scenarios.

Reply:
Got it, thank you for clarifying 🙏